src/test/regress/upgrade/config.py

```diff
@@ -1,68 +1,232 @@
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 1
+
+
+class CitusSingleNodeSingleShardClusterConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 0
+        self.new_settings = {"citus.shard_count": 1}
+        self.settings.update(self.new_settings)
+
+
+class CitusShardReplicationFactorClusterConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.shard_replication_factor": 2}
+        self.settings.update(self.new_settings)
+
+
+class CitusNoLocalExecutionClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.enable_local_execution": False}
+        self.settings.update(self.new_settings)
+
+
+class CitusComplexClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {
+            "citus.enable_local_execution": False,
+            "citus.multi_shard_commit_protocol": "1pc",
+            "citus.multi_shard_modify_mode": "sequential",
+            "citus.prevent_incomplete_connection_establishment": False,
         }
+        self.settings.update(self.new_settings)
 
 
-USER = 'postgres'
-DBNAME = 'postgres'
+class CitusSingleShardClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.shard_count": 1}
+        self.settings.update(self.new_settings)
+
+
+class CitusNonMxClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.is_mx = False
+
+
+class CitusManyShardsClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.shard_count": 500}
+        self.settings.update(self.new_settings)
 
-COORDINATOR_NAME = 'coordinator'
-WORKER1 = 'worker1'
-WORKER2 = 'worker2'
-NODE_NAMES = [COORDINATOR_NAME, WORKER1, WORKER2]
-COORDINATOR_PORT = 57635
-WORKER1PORT = 57636
-WORKER2PORT = 57637
 
-WORKER_PORTS = [WORKER1PORT, WORKER2PORT]
-NODE_PORTS = {
-    COORDINATOR_NAME: COORDINATOR_PORT,
-    WORKER1: WORKER1PORT,
-    WORKER2: WORKER2PORT,
-}
+class CitusSingleNodeSingleConnectionClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.max_adaptive_executor_pool_size": 1}
+        self.settings.update(self.new_settings)
+
+
+class CitusSingleNodeSingleSharedPoolSizeClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.max_shared_pool_size": 1}
```

Comment:
I think citus.max_shared_pool_size  = 1 is too aggressive. The maintenance daemon uses/caches 1 connection, which prevents other queries to succeed

Reply:
Interesting, do you know why we don't get any failures currently?

Reply:
Oh I saw this comment now: https://github.com/citusdata/citus/pull/5316#discussion_r720075711